### DISCUSSION TITLE
Gem::GemNotFoundException: can't find gem bundler (= 2.4.13) with executable bundle

### DISCUSSION THREAD

**Discussion by ibis-git** [ruby/rubygems#6712] [Q&A] [answered]
Gem::GemNotFoundException: can't find gem bundler (= 2.4.13) with executable bundle
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/rubygems.rb:263:in `find_spec_for_exe'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/rubygems.rb:243:in `bin_path'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/rubygems_integration.rb:178:in `bin_path'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/shared_helpers.rb:282:in `set_bundle_variables'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/shared_helpers.rb:76:in `set_bundle_environment'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/cli/exec.rb:20:in `run'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/cli.rb:492:in `exec'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/cli.rb:34:in `dispatch'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/cli.rb:28:in `start'
  /opt/homebrew/Cellar/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.10/exe/bundle:45:in `block in <top (required)>'
  /opt/homebrew/lib/ruby/site_ruby/3.2.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
  /opt/homebrew/Cellar/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.10/exe/bundle:33:in `<top (required)>'
  /opt/homebrew/opt/ruby/bin/bundle:25:in `load'
  /opt/homebrew/opt/ruby/bin/bundle:25:in `<main>'

```

## Environment

```
Bundler       2.4.13
  Platforms   ruby, arm64-darwin-22
Ruby          3.2.2p53 (2023-03-30 revision e51014f9c05aa65cbf203442d37fef7c12390015) [arm64-darwin-22]
  Full Path   /opt/homebrew/opt/ruby/bin/ruby
  Config Dir  /opt/homebrew/Cellar/ruby/3.2.2/etc
RubyGems      3.4.13
  Gem Home    /opt/homebrew/lib/ruby/gems/3.2.0
  Gem Path    /Users/kohrin/.gem/ruby/3.2.0:/opt/homebrew/lib/ruby/gems/3.2.0:/opt/homebrew/Cellar/ruby/3.2.2/lib/ruby/gems/3.2.0
  User Home   /Users/kohrin
  User Path   /Users/kohrin/.gem/ruby/3.2.0
  Bin Dir     /opt/homebrew/lib/ruby/gems/3.2.0/bin
Tools         
  Git         2.40.1
  RVM         not installed
  rbenv       not installed
  chruby      not installed
Gem.ruby      /opt/homebrew/opt/ruby/bin/ruby
bundle #!     /opt/homebrew/Cellar/ruby/3.2.2/bin/ruby

```

## Bundler Build Metadata

```
Built At          2023-05-10
Git SHA           26eb456c6c
Released Version  true
```

## Bundler settings

```
gemfile
  Set via BUNDLE_GEMFILE: "/opt/homebrew/Cellar/shopify-cli/3.46.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile"
path
  Set for your local app (/opt/homebrew/Cellar/shopify-cli/3.46.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/.bundle/config): "/Users/kohrin/Library/Caches/shopify-gems-nodejs"
without
  Set for your local app (/opt/homebrew/Cellar/shopify-cli/3.46.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/.bundle/config): [:development, :test]
```

## Gemfile

### ../../../../../../opt/homebrew/Cellar/shopify-cli/3.46.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile

```ruby
# NOTE: These are development-only dependencies
source "https://rubygems.org"

gem "bugsnag", "~> 6.22"
gem "listen", "~> 3.7.0"
gem "theme-check", "~> 1.15.0"
gem "nokogiri", "~> 1.14.0"

# None of these can actually be used in a development copy of dev
# They are all for CI and tests
# `dev` uses no gems
group :development, :test do
  gem "pry-byebug"
  gem "byebug"
  gem "rubocop-shopify", require: false
  gem "rubocop-minitest", require: false
  gem "rubocop-rake", require: false
  gem "iniparse", "~> 1.5"
  gem "colorize", "~> 0.8.1"
  gem "bundler", ">= 2.3.11"
  gem "rake", "~> 12.3", ">= 12.3.3"
  gem "minitest", "~> 5.0"
end

group :test do
  gem "mocha", require: false
  gem "minitest-reporters", require: false
  gem "minitest-fail-fast", require: false
  gem "fakefs", ">= 1.0", require: false
  gem "webmock", require: false
  gem "timecop", require: false
  gem "rack", require: false
  gem "cucumber", "~> 7.0", require: false
end
```

### ../../../../../../opt/homebrew/Cellar/shopify-cli/3.46.1/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    addressable (2.8.4)
      public_suffix (>= 2.0.2, < 6.0)
    ansi (1.5.0)
    ast (2.4.2)
    bugsnag (6.25.2)
      concurrent-ruby (~> 1.0)
    builder (3.2.4)
    byebug (11.1.3)
    coderay (1.1.3)
    colorize (0.8.1)
    concurrent-ruby (1.2.2)
    crack (0.4.5)
      rexml
    cucumber (7.1.0)
      builder (~> 3.2, >= 3.2.4)
      cucumber-core (~> 10.1, >= 10.1.0)
      cucumber-create-meta (~> 6.0, >= 6.0.1)
      cucumber-cucumber-expressions (~> 14.0, >= 14.0.0)
      cucumber-gherkin (~> 22.0, >= 22.0.0)
      cucumber-html-formatter (~> 17.0, >= 17.0.0)
      cucumber-messages (~> 17.1, >= 17.1.1)
      cucumber-wire (~> 6.2, >= 6.2.0)
      diff-lcs (~> 1.4, >= 1.4.4)
      mime-types (~> 3.3, >= 3.3.1)
      multi_test (~> 0.1, >= 0.1.2)
      sys-uname (~> 1.2, >= 1.2.2)
    cucumber-core (10.1.1)
      cucumber-gherkin (~> 22.0, >= 22.0.0)
      cucumber-messages (~> 17.1, >= 17.1.1)
      cucumber-tag-expressions (~> 4.1, >= 4.1.0)
    cucumber-create-meta (6.0.4)
      cucumber-messages (~> 17.1, >= 17.1.1)
      sys-uname (~> 1.2, >= 1.2.2)
    cucumber-cucumber-expressions (14.0.0)
    cucumber-gherkin (22.0.0)
      cucumber-messages (~> 17.1, >= 17.1.1)
    cucumber-html-formatter (17.0.0)
      cucumber-messages (~> 17.1, >= 17.1.0)
    cucumber-messages (17.1.1)
    cucumber-tag-expressions (4.1.0)
    cucumber-wire (6.2.1)
      cucumber-core (~> 10.1, >= 10.1.0)
      cucumber-cucumber-expressions (~> 14.0, >= 14.0.0)
    diff-lcs (1.5.0)
    fakefs (2.5.0)
    ffi (1.15.5)
    hashdiff (1.0.1)
    iniparse (1.5.0)
    json (2.6.3)
    liquid (5.4.0)
    listen (3.7.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    method_source (1.0.0)
    mime-types (3.4.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2023.0218.1)
    minitest (5.18.0)
    minitest-fail-fast (0.1.0)
      minitest (~> 5)
    minitest-reporters (1.6.0)
      ansi
      builder
      minitest (>= 5.0)
      ruby-progressbar
    mocha (2.0.2)
      ruby2_keywords (>= 0.0.5)
    multi_test (0.1.2)
    nokogiri (1.14.5-arm64-darwin)
      racc (~> 1.4)
    parallel (1.23.0)
    parser (3.2.2.1)
      ast (~> 2.4.1)
    pry (0.14.2)
      coderay (~> 1.1)
      method_source (~> 1.0)
    pry-byebug (3.10.1)
      byebug (~> 11.0)
      pry (>= 0.13, < 0.15)
    public_suffix (5.0.1)
    racc (1.6.2)
    rack (3.0.7)
    rainbow (3.1.1)
    rake (12.3.3)
    rb-fsevent (0.11.2)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    regexp_parser (2.8.0)
    rexml (3.2.5)
    rubocop (1.51.0)
      json (~> 2.3)
      parallel (~> 1.10)
      parser (>= 3.2.0.0)
      rainbow (>= 2.2.2, < 4.0)
      regexp_parser (>= 1.8, < 3.0)
      rexml (>= 3.2.5, < 4.0)
      rubocop-ast (>= 1.28.0, < 2.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 2.4.0, < 3.0)
    rubocop-ast (1.28.1)
      parser (>= 3.2.1.0)
    rubocop-minitest (0.31.0)
      rubocop (>= 1.39, < 2.0)
    rubocop-rake (0.6.0)
      rubocop (~> 1.0)
    rubocop-shopify (2.13.0)
      rubocop (~> 1.50)
    ruby-progressbar (1.13.0)
    ruby2_keywords (0.0.5)
    sys-uname (1.2.3)
      ffi (~> 1.1)
    theme-check (1.15.0)
      liquid (>= 5.4.0)
      nokogiri (>= 1.12)
      parser (~> 3)
    timecop (0.9.6)
    unicode-display_width (2.4.2)
    webmock (3.18.1)
      addressable (>= 2.8.0)
      crack (>= 0.3.2)
      hashdiff (>= 0.4.0, < 2.0.0)

PLATFORMS
  arm64-darwin-22

DEPENDENCIES
  bugsnag (~> 6.22)
  bundler (>= 2.3.11)
  byebug
  colorize (~> 0.8.1)
  cucumber (~> 7.0)
  fakefs (>= 1.0)
  iniparse (~> 1.5)
  listen (~> 3.7.0)
  minitest (~> 5.0)
  minitest-fail-fast
  minitest-reporters
  mocha
  nokogiri (~> 1.14.0)
  pry-byebug
  rack
  rake (~> 12.3, >= 12.3.3)
  rubocop-minitest
  rubocop-rake
  rubocop-shopify
  theme-check (~> 1.15.0)
  timecop
  webmock

BUNDLED WITH
   2.4.13
```

**Comment by altivi** [comment, marked as answer]
@ibis-git The error message indicates that Bundler version 2.4.13 is not found or executable.

To resolve this issue, you can try the following steps:

- Verify the installed version of Bundler by running `gem list bundler` in your terminal. Make sure that version 2.4.13 is listed.

- If Bundler version 2.4.13 is not installed, you can install it by running `gem install bundler -v 2.4.13`.

- If Bundler version 2.4.13 is already installed, you can try uninstalling it first by running `gem uninstall bundler -v 2.4.13` and then reinstalling it using the previous command.

- After ensuring that Bundler version 2.4.13 is installed, you can try running your desired command again.
If the issue persists, it could be due to an incompatible setup or a configuration problem. In that case, please provide more details about the specific command you're running and any additional error messages you're receiving.

Everything should be done in the directory where Gemfile is also present.

If it was helpful, please mark as answer.

**Comment by deivid-rodriguez** [comment]
Closing since this was answered in detail!
